### PR TITLE
Introduce cta-landing-page-type for story ads

### PR DESCRIPTION
**Pull request by calebcordry**
This optional meta tag indicates the type of landing page for an amp story ad.

3 choices: 
`<meta name="amp-cta-landing-page-type" content="NONAMP">`
`<meta name="amp-cta-landing-page-type" content="AMP">`
`<meta name="amp-cta-landing-page-type" content="STORY">`

### CODE REVIEW

File: validator/validator-main.protoascii [hunk 1]
```diff
@@ -867,6 +867,25 @@ tags: {
...
+# Specifies the Single Page Story Ad landing page type
+tags: {
+  html_format: AMP4ADS
+  tag_name: "META"
+  spec_name: "meta name=amp-cta-landing-page-type"
+  mandatory_parent: "HEAD"
+  unique: true
+  attrs: {
+    name: "content"
+    mandatory_oneof: ['AMP', 'NOAMP', 'STORY']
```
Comment (twifkak) [inline validator/validator-main.protoascii]: I think this needs to be in double-quotes, per the other examples in this file.

File: validator/validator-main.protoascii [hunk 2]
```diff
@@ -867,6 +867,25 @@ tags: {
...
+  }
+  attrs: {
+    name: "name"
+    mandatory: true
+    value_casei: "amp-cta-landing-page-type"
+    dispatch_key: NAME_VALUE_DISPATCH
```
Comment (twifkak) [inline validator/validator-main.protoascii]: I don't understand the comments in validator.proto well enough to know if this is needed. Can you add a positive test and a negative test for this change?

File: validator/validator-main.protoascii (hunk 1 above)
Comment (calebcordry) [reply validator/validator-main.protoascii]: Done.

File: validator/validator-main.protoascii [hunk 3]
```diff
@@ -26,7 +26,7 @@ min_validator_revision_required: 348
 # newer versions of the spec file. This is currently a Google internal
 # mechanism, validator.js does not use this facility. However, any
 # change to this file (validator-main.js) requires updating this revision id.
-spec_file_revision: 790
+spec_file_revision: 791
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: we don't need to update this here.

File: validator/validator-main.protoascii [hunk 4]
```diff
@@ -867,6 +867,25 @@ tags: {
...
+# Specifies the Single Page Story Ad landing page type
+tags: {
+  html_format: AMP4ADS
+  tag_name: "META"
+  spec_name: "meta name=amp-cta-landing-page-type"
+  mandatory_parent: "HEAD"
+  unique: true
+  attrs: {
+    name: "content"
+    mandatory_oneof: "['AMP', 'NONAMP', 'STORY']"
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: these should be lowercase.

File: validator/validator-main.protoascii (hunk 2 above)
Comment (honeybadgerdontcare) [reply validator/validator-main.protoascii]: this is necessary. but tests are good as well

Comment (calebcordry) [reply validator/validator-main.protoascii]: @honeybadgerdontcare just for my knowledge could you clarify what this is actually doing?

Comment (honeybadgerdontcare) [reply validator/validator-main.protoascii]: I guess the comment isn't clear enough in [validator.proto](https://github.com/ampproject/amphtml/blob/master/validator/validator.proto#L136).

In essence it is telling the validator that this tag spec for this tag, when it has this attribute with this attribute name and attribute value pair should be evaluated first before any other tagspec with this tag and this attribute name. Only one tagspec for this tag with this attribute name and attribute value may have `dispatch_key: NAME_VALUE_DISPATCH`. It's giving a hint to the validator that it's the more relevant tagspec.

Does that help?

Comment (calebcordry) [reply validator/validator-main.protoascii]: Yes, thanks. Added some tests.

File: validator/validator-main.protoascii (hunk 3 above)
Comment (calebcordry) [reply validator/validator-main.protoascii]: Done.

File: validator/validator-main.protoascii (hunk 4 above)
Comment (calebcordry) [reply validator/validator-main.protoascii]: Done.

File: validator/testdata/amp4ads_feature_tests/amp_story_ad_errors.out [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+>>     ^~~~~~~~~
+amp4ads_feature_tests/amp_story_ad_errors.html:28:4 The attribute 'name' in tag 'meta name= and content=' is set to the invalid value 'amp-cta'. [DISALLOWED_HTML]
+|
+|      <!-- Should fail due to no `content=`. -->
+|      <meta name="amp-cta-type">
+>>     ^~~~~~~~~
+amp4ads_feature_tests/amp_story_ad_errors.html:31:4 The mandatory attribute 'content' is missing in tag 'meta name=amp-cta-type'. [DISALLOWED_HTML]
+|
+|      <!-- Should fail due to invalid content choice. -->
+|      <meta name="amp-cta-landing-page-type" content="GAME">
```
Comment (calebcordry) [inline validator/testdata/amp4ads_feature_tests/amp_story_ad_errors.out]: @honeybadgerdontcare @twifkak any idea why this isn't failing?

**Comment by twifkak** [comment]
Hi, y'all. I'll be on vacation next 2 weeks. @honeybadgerdontcare speaks for me.

File: validator/testdata/amp4ads_feature_tests/amp_story_ad_errors.out (hunk 5 above)
Comment (calebcordry) [reply validator/testdata/amp4ads_feature_tests/amp_story_ad_errors.out]: Done.

**Pull request merged**